docs/compiler-exceptions.rst

```diff
@@ -0,0 +1,276 @@
+
+.. py:exception:: NonPayableViolationException
+
+    .. code-block:: python3
+
+        @private
+        def _foo():
+            bar: uint256 = msg.value
+       
+    Raised when attempting to access ``msg.value`` from within a private function.
+
+.. py:exception:: ParserException
+
+    .. code-block:: python3
+
+        @private
+        def foo(a: address = msg.sender):
+            pass
+        vyper.exceptions.ParserException: line 3:21 msg.sender not allowed in private functions.
+             2 @private
+        ---> 3 def foo(a: address = msg.sender): pass
+        ----------------------------^
+
+    See ``PythonSyntaxException``.  This error is not commonly used and typically refers to an error of one of the other types.
+
+.. py:exception:: PythonSyntaxException
+
+    .. code-block:: python3
+
+        >>> while True print('Hello world')
+            File "<stdin>", line 1
+            while True print('Hello world')
+                   ^
+            SyntaxError: invalid syntax
+     
+    This exception is raised due to a python based syntax exception.  It is also known as a parser exception.  In this case there is a colon missing after the word print.  See `Python Errors and Exceptions <https://docs.python.org/3/tutorial/errors.html>`_ for more details.
+
+.. py:exception:: StructureException
+
+    .. code-block:: python3
+
+        # Transfer funds to beneficiary
+        send(self.beneficiary, self.highestBid)
+
+        '''
+
+    The inclusion of punctuation that is syntactically incorrect and not a part of the normal vyper flow will throw a 
+    ``StructureException``.
+
+    .. code-block:: bash
+
+        vyper.exceptions.StructureException: line 181:0 Invalid top-level statement
+             180
+        ---> 181 '''
+        ---------^
+             182  
+    The terminal displays the line number and points directly to the problem.
+
+.. py:exception:: SyntaxException
+
+    .. code-block:: python3
+
+        struct Bid:
+            blindedBid bytes32
+            deposit: wei_value
+
+    A syntax error is thrown in the declaration of this ``struct`` variable.
+
+    .. code-block:: bash
+
+    $ vyper blind_auction.vy
+        vyper.exceptions.PythonSyntaxException: line 4:20 SyntaxError: invalid syntax
+             3 struct Bid:
+        ---> 4   blindedBid bytes32
+        ---------------------------^
+             5   deposit: wei_value
+
+    The terminal output of a syntax error will generally show exactly where it happened.  In this case there is a semi
+    colon missing after ``blindedBid`` in the declaration of the struct.
+
+.. py:exception:: TypeMismatchException
+
+.. code-block:: python3
+
+    @public
+    def foo():
+    a: uint256 = 42
+    b: bytes32 = a
+    vyper.exceptions.TypeMismatchException: line 4:4 Invalid type, expected: bytes32
```

Comment:
Can you split this `code-block` in two, starting at line 210? And then move the text at line 215 in between the two blocks, in order to maintain a consistent style with the other docs.

Reply:
This is still unresolved.